implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -141,7 +144,7 @@ private void checkNameClaims(JwtContext jwtContext) throws InvalidJwtException {
 
     private String findSubject(JWTAuthContextInfo authContextInfo, JwtClaims claimsSet) {
         if (authContextInfo.getSubjectPath() != null) {
-            final String[] pathSegments = authContextInfo.getSubjectPath().split("/");
+            final String[] pathSegments = CLAIM_PATH_PATTERN.split(authContextInfo.getSubjectPath());
```

Comment:
Would it be better to do the `split` when the configuration is loaded rather than per request?

Reply:
@MikeEdgar 
We'll need to deprecate then the group and sub path setters/getters in `JWTAuthContextInfo` and return the arrays and 'lose' the complete path for the log purposes when a given token does not match etc and I'm not 100% sure it is worth it, as it is likely a good number of the custom paths just point to a top level claim which is different to `groups` and such we'll be optimizing for like 20%. What do you think about a minor optimization for this what appears to be the typical case: split only if a '/' character is contained - it would allow us just to get the claim path value as is in 80% ?
Thanks

Reply:
I agree that most apps will use this to identify another top-level claim. Checking for the `/` delimiter first is a good idea.

Reply:
Sounds good. So indeed, the custom claims case in itself is probably 30% max, and within that percentage with this simple check we can limit the pattern split to a likely 5% or 10% max case. let me start with it and I'll have no problems with going the deprecation route later if there be some evidence the split is used quite a lot